fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -1346,6 +1352,244 @@ def assert_state(self, state: Union[TrainingState, List[TrainingState]]) -> None
                 traceback.print_stack()
             raise ValueError(msg)
 
+    # Optim State dict functions
```

Comment:
I considered moving these to a separate `FSDPOptimizerMixin` in `fsdp_optimizer_utils.py`, but decided it wasn't really a mixin since it depends heavily on FSDP.